Lib/pdb.py

```diff
@@ -820,8 +888,20 @@ def default(self, line):
                 sys.stdout = save_stdout
                 sys.stdin = save_stdin
                 sys.displayhook = save_displayhook
-        except:
-            self._error_exc()
+        except Exception as e:
+            # Maybe it's an await expression/statement
+            if (
+                self.async_shim_frame is not None
+                and isinstance(e, SyntaxError)
+                and e.msg == "'await' outside function"
+            ):
+                try:
+                    self._exec_await(buffer, globals, locals)
```

Comment:
I think this should be executed without the exception from line 891 in the context